OneCommunityGlobal/HighestGoodNetworkApp
# Wanting make 5 letter codes editable only by admin&owner

# Description
<img width="882" alt="截屏2023-10-20 下午1 03 00" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/bef20af5-5117-4735-9348-6189d81cffb6">


## Related PRS (if any):
…

## Main changes explained:
- Update WeeklySummariesReport.jsx file for Owner/Admin/whoever is given "edit team code" permission to edit 5-letter team code in 'Weekly Summaries Report page'
- Update UserProfile.jsx file for Owner/Admin/whoever is given "edit team code" permission to edit 5-letter team code in profile's teams tab
…

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as owner/admin user
        a. go to Reports -> Weekly Summaries Report
        b. verify that the 5-letter team code is editable
        c. go to User Profile -> Teams tab
        d. verify that the 5-letter team code is editable
4. edit a team code, refresh the page, and check the edited team code is saved
5. log as any role EXCEPT admin/owner(like volunteer, manager, mentor, etc.)
        a. go to Reports -> Weekly Summaries Report
        b. verify that the 5-letter team code is NOT editable
        c. go to User Profile -> Teams tab
        d. verify that the 5-letter team code is NOT editable


## Screenshots or videos of changes:
**editable team code example:**
![截屏2023-10-20 下午1 11 24](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/e594c116-42a4-4e8f-b9d1-7914b831fe6d)

![截屏2023-10-20 下午1 45 17](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/10a1ff4d-5f99-4853-b033-03b710626976)


**uneditable team code example:**
![截屏2023-10-20 下午1 17 32](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/eb455ec6-05f4-4859-a67f-c4d45c443c33)

![截屏2023-10-20 下午1 21 32](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105323719/4a4e3de3-9958-41d2-82a9-ac63af0d85bc)
